test/lib/y2firewall/clients/auto_test.rb

```diff
@@ -33,52 +33,63 @@
     allow(subject).to receive(:importer).and_return(importer)
   end
 
-  describe "#general_summary" do
-    it "creates a text which includes firewalld overview" do
-      expect(firewalld).to receive(:running?).and_return(true)
-      expect(firewalld).to receive(:enabled?).and_return(true)
-      expect(firewalld).to receive(:default_zone).and_return("public")
-
-      summary = subject.send(:general_summary)
+  describe "#zone_summary" do
+    it "empty zone returns empty description" do
+      summary = subject.send(:zone_summary, Y2Firewall::Firewalld::Zone.new(name: "test_zone"))
 
-      expect(summary).to match(/Running/)
-      expect(summary).to match(/Enabled/)
-      expect(summary).to match(/Default zone/)
-      expect(summary).to match(/Defined zones/)
+      expect(summary).to be_empty
     end
   end
 
-  describe "#zone_summary" do
-    RELATIONS = {
-      interfaces: ["eth0", "eth1"],
-      services:   ["ssh", "ftp"],
-      protocols:  ["udp", "tcp"],
-      ports:      ["80"]
-    }.freeze
+  describe "#summary" do
+    context "when firewalld is not installed" do
+      before(:each) do
+        allow(firewalld).to receive(:installed?).and_return(false)
+      end
+
+      it "reports when firewalld is not available" do
+        expect(subject.summary).to match(/not available/)
+      end
+    end
 
-    def define_zone(name: nil, relation: nil, values: nil)
-      return nil if name.nil? || name.empty?
-      return nil if !Y2Firewall::Firewalld::Zone.instance_methods(false).include?(relation)
+    context "when firewalld is installed" do
+      let(:relations_stub) do
+        {
+          interfaces: ["eth0", "eth1"],
+          services:   ["ssh", "ftp"],
+          protocols:  ["udp", "tcp"],
+          ports:      ["80"]
+        }
+      end
 
-      zone = Y2Firewall::Firewalld::Zone.new(name: name)
-      expect(zone).to receive(relation).and_return(values)
+      before(:each) do
+        zones = []
+        relations_stub.each_pair do |relation, values|
```

Comment:
NP: You could even declare the zone with a let and then iterate over the stubs, and the relations can be assigned directly without mocking. Also as we are just checking that the summary contains the values, multiple zones are not necessary at all.

But I'm fine as it is.
But I'm fine as it is.